src/app/user/user-account/user-account.component.ts

```diff
@@ -16,22 +16,28 @@
  */
 
 import { Component, OnInit } from '@angular/core';
-import { ChangePasswordService } from '../../service/change-password.service';
+import { User } from '../user';
+import {SessionService} from '../../session/session.service';
+
 @Component({
   selector: 'app-user-account',
   templateUrl: './user-account.component.html',
   styleUrls: ['./user-account.component.css']
 })
 export class UserAccountComponent implements OnInit {
 
-  constructor( private ChangePasswordService: ChangePasswordService ) { }
+  private user: User;
+
+  constructor( private sessionService: SessionService ) { }
 
   ngOnInit() {
+    this.user = this.sessionService.getCurrentUser();
   }
 
   changePassword( pwd , rpwd ) {
     if ( pwd === rpwd ) {
-      this.ChangePasswordService.setNewPassword( pwd );
+      // TODO: update the password for this remote database user first and deny the password change if that fails
+      this.user.setNewPassword( pwd );
```

Comment:
Hmm that's a good question. Especially when we are going to provide offline functionality in future this get's pretty tricky. I think we should postpone this until we figure out how we might seperate local and remote session handling, like proposed in #78 and while we're on that, we should rethink this whole login system, too.